NETWORKDAYS function and string support DATEDIF

scytacki opened the pull request:
## Context

As a Grist user, I want to know how many working days are between two dates, so I can create timesheet documents which compute hours per working day.

In most spreadhseets this is supported by the NETWORKDAYS (net work days) function. In Airtable it is supported by WORKDAY_DIFF. Some spreadsheets also provide a NETWORKDAYS.INTL which allows defining what days are the "weekend".

This new function is not necessary because a user can do this in Python, but since it is a common spreadsheet function it seems worth adding it to the core. 

## Proposed solution

Add the NETWORKDAYS function to Grist. 

While adding this I saw that the argument parsing overlapped with the DATEDIF function so I added a `_make_date` helper function which handles `datetime.datetime`, `datetime.date`, and `str` values. So now DATEDIF also supports string arguments.

Potential Backward Compatibility issue:
If a document is relying on DATEDIF throwing a `TypeError` when it is passed a string that document will no longer behave the same way.

Further Improvements:
- The current implementation of NETWORKDAYS is simple and concise but probably not as efficient as possible with large ranges. It is iterating over every date in the range rather than going week by week. Without holidays the week by week approach is definitely faster, but with holidays it might not make much of a difference. If efficiency is a concern then we probably should figure out how to profile different implementations to see which one is best under a variety of conditions.
- This implementation has hardcoded weekends. Given the international nature of Grist, perhaps it should support customizing them.
- The name NETWORKDAYS is confusing (at least to me). Airtable addressed this, perhaps Grist should too. I don't know if there are any guidelines about how close the Grist formula functions should match standard spreadsheet functions.
- Perhaps the documentation of the DATEDIF was extended too much.

## Related issues

This is a new feature, and I didn't discuss it in an issue first. Please, let me know if that is still necessary.

## Has this been tested?

<!-- Put an `x` in the box that applies: -->

- [x] 👍 yes, I added tests to the test suite
- [ ] 💭 no, because this PR is a draft and still needs work
- [ ] 🙅 no, because this is not relevant here
- [ ] 🙋 no, because I need help <!-- Detail how we can help you -->

The two updated functions have examples added to their doc-comments which are tested by test_functions.py.

## Notes

This is my first time using many of these Python features, so hopefully I didn't miss something.

dsagal reviewed (commented):
I also have a review request -- to add a couple more test cases.

Thank you for this function, this is great work!

scytacki commented:
@dsagal and @georgegevoian thanks for reviewing this. I believe I've addressed your comments:
- It now returns a negative number when the start_date is after the end_date
- There are more test cases, including the ones from the google sheets documentation. One of which is actually incorrect when tested in google sheets itself (as you also identified).
